### DISCUSSION TITLE
Date formatting in filter web part

### DISCUSSION THREAD

**Discussion by upcreek** [microsoft-search/pnp-modern-search#2982] [Q&A] [answered]
Hi all, I've been working on formatting dates in the search web parts..so far I've been able to format in the results web part and part of the filter web part using some simple handlebars edits in the templates. But one part of the filter web part is eluding me:

![image](https://github.com/microsoft-search/pnp-modern-search/assets/37710523/eea2ab22-0a23-48f5-98d6-9b351529d725)

As the page is loading I can see the date is formatted how I want it but the reverts to the full format when the page finishes loading.

Anyone know how I can format this date? I'm using the checkbox filter template. Is it possible to get that deep into the template or do I need to do something completely custom? Thanks.

**Comment by wobba** [comment]
Can you share a bit of your setup?

**Reply by wobba** [reply]
Curious on the exact web part configuration, what property you use for refiner, etc. What are steps needed to repro what you are experiencing?

**Reply by upcreek** [reply]
Oh of course! I'm using a search filter web part and a search results web part. Here's my filter:

![image](https://github.com/microsoft-search/pnp-modern-search/assets/37710523/e5867f44-fb86-4f1b-89a0-287bf2a1f923)

I'm using one of the reserved refinable managed properties on the site collection and I have a slot created for this property in the search results and I'm using handlebars in both the results and filter layouts to format the date. Specifically, in the filters:

```
<div class="filter--values-list">
  {{#each filter.values}}
	  <div title="{{getDate name 'YYYY-MM-DD'}} {{#if ../showCount}}({{count}}){{/if}}" class="filter--value">
		  <pnp-filtercheckbox 
			  data-instance-id="{{@root.instanceId}}" 
			  data-filter-name="{{filter.filterName}}" 
			  data-name="{{getDate name 'YYYY-MM-DD'}}" 
			  data-value="{{getDate value 'YYYY-MM-DD'}}" 
			  data-selected="{{selected}}" 
			  data-disabled="{{disabled}}" 
			  data-count="{{count}}"
			  data-is-multi="{{../isMulti}}"
			  data-theme-variant="{{JSONstringify @root.theme}}"
		  >
		  </pnp-filtercheckbox>
		  {{#if ../showCount}}
			  <span data-ui-test-id="filterCount">({{count}})</span>
		  {{/if}}
	  </div>
  {{/each}}
  </div>
```



Let me know if you need more info.

**Reply by wobba** [reply, marked as answer]
I would recommend using RefinableDate properties for date type values, and then use one of the date templates. That said, you should not format the date in the data-value, as then you would pass the wrong value down when filtering (aside from the UX issue you report). Could be a repaint issue of sorts.
